ephios/api/views/events.py

```diff
@@ -119,29 +67,35 @@ class EventViewSet(viewsets.ReadOnlyModelViewSet):
     )
 
 
-class ParticipationSerializer(serializers.ModelSerializer):
-    event_title = serializers.CharField(source="shift.event.title")
-    state = ChoiceDisplayField(choices=AbstractParticipation.States.choices)
-    duration = serializers.SerializerMethodField()
+class UserinfoParticipationViewSet(viewsets.ReadOnlyModelViewSet):
+    serializer_class = UserinfoParticipationSerializer
+    permission_classes = [ViewUserModelObjectPermissions, IsAuthenticatedOrTokenHasScope]
+    filter_backends = [UserinfoParticipationPermissionFilter, DjangoFilterBackend]
+    filterset_class = ParticipationFilterSet
+    required_scopes = ["CONFIDENTIAL_READ"]
+
+    queryset = (
+        AbstractParticipation.objects.all()
+        .select_related("shift", "shift__event", "shift__event__type")
+        .order_by("id")
+    )
 
-    class Meta:
-        model = LocalParticipation
-        fields = [
-            "id",
-            "shift",
-            "event_title",
-            "state",
-            "comment",
-            "start_time",
-            "end_time",
-            "duration",
-            "structure_data",
-        ]
 
-    def build_unknown_field(self, field_name, model_class):
-        if field_name in {"start_time", "end_time"}:
-            return self.build_property_field(field_name, model_class)
-        return super().build_unknown_field(field_name, model_class)
+class ParticipationViewSet(UserinfoParticipationViewSet):
+    """
+    Remove information that would not be visible in the web version,
+    i.e. no email and date of birth of participants
+    """
 
-    def get_duration(self, obj):
-        return (obj.end_time - obj.start_time).total_seconds()
+    serializer_class = ParticipationSerializer
+    filter_backends = [ParticipationPermissionFilter, DjangoFilterBackend]
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["CONFIDENTIAL_READ"]
```

Comment:
shouldn't that be PUBLIC_READ for the redacted version?

Reply:
It may be visible for logged in users, but (as per our docs) the scopes consider any people-related data as confidential.